plutus-core/executables/plutus/AnyProgram/With.hs

```diff
@@ -1,28 +1,41 @@
+{-# LANGUAGE AllowAmbiguousTypes      #-}
 {-# LANGUAGE DataKinds                #-}
 {-# LANGUAGE FlexibleInstances        #-}
 {-# LANGUAGE MultiParamTypeClasses    #-}
 {-# LANGUAGE PolyKinds                #-}
+{-# LANGUAGE QuantifiedConstraints    #-}
 {-# LANGUAGE RankNTypes               #-}
 {-# LANGUAGE StandaloneKindSignatures #-}
+{-# LANGUAGE TypeOperators            #-}
+{-# LANGUAGE UndecidableInstances     #-}
 {-# LANGUAGE UndecidableSuperClasses  #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
 -- | BOILERPLATE needed to support Hasochism.
 -- See <https://homepages.inf.ed.ac.uk/slindley/papers/hasochism.pdf>
 module AnyProgram.With where
 
+import PlutusCore qualified as PLC
 import PlutusCore.Data qualified as PLC
 import PlutusCore.Pretty as PLC
+import PlutusIR qualified as PIR
 import UntypedPlutusCore qualified as UPLC
 
 import Data.Kind (Constraint)
-import Flat
 import Types
 
 -- for: (typeclass `compose` type)
 type ComposeC :: forall a b. (b -> Constraint) -> (a -> b) -> a -> Constraint
 class    constr (f x) => ComposeC constr f x
 instance constr (f x) => ComposeC constr f x
 
+type UnitC :: forall a. a -> Constraint
+class    UnitC x
+instance UnitC x
+
+type AndC :: forall a. (a -> Constraint) -> (a -> Constraint) -> a -> Constraint
+class    (constr1 x, constr2 x) => AndC constr1 constr2 x
+instance (constr1 x, constr2 x) => AndC constr1 constr2 x
+
 withN :: forall constr s r
```

Comment:
I hate these names BTW, `withName` wouldn't be much longer and it would be much more readable.